classes/submission/Repository.php

```diff
@@ -576,4 +608,34 @@ protected function getCurrentPublicationIdByPublications(Submission $submission)
 
         return $newCurrentPublicationId ?? $submission->getData('currentPublicationId');
     }
+
+    /**
+     * Checks if this user is granted access to preview
+     * based on their roles in the context (i.e. Manager, Editor, etc).
+     *
+     * @param $user User
+     * @param $submission Submission
+     *
+     * @return bool
+     */
+    protected function _roleCanPreview(User $user, Submission $submission): bool
+    {
+        $roleDao = DAORegistry::getDAO('RoleDAO'); /* @var $roleDao RoleDAO */
+        if ($user && $submission) {
+            $subscriptionAssumedRoles = [
+                ROLE_ID_MANAGER,
+                ROLE_ID_SUB_EDITOR,
+                ROLE_ID_ASSISTANT,
+                ROLE_ID_SUBSCRIPTION_MANAGER
```

Comment:
This role ID is only defined in pkp-lib, so I suppose this would probably cause a fatal error (in recent versions of PHP) in OMP and OPS.

Reply:
So `Role::ROLE_ID_MANAGER` instead, right?

Reply:
added `Role::`

Reply:
No, `ROLE_ID_SUBSCRIPTION_MANAGER` is an OJS-specific role (and not the same thing as `ROLE_ID_MANAGER`). This would need to be included only in OJS, so I suppose you'll want to use the OJS subclass to augment the pkp-lib list with `ROLE_ID_SUBSCRIPTION_MANAGER`.